clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -472,8 +472,28 @@ void SYCLToolChain::TranslateTargetOpt(const llvm::opt::ArgList &Args,
   }
 }
 
+static void addImpliedArgs(const llvm::Triple &Triple,
+                           const llvm::opt::ArgList &Args,
+                           llvm::opt::ArgStringList &CmdArgs) {
+  // Current implied args are for debug information and disabling of
+  // optimizations.
+  // TODO: Add support for other architectures (gen, x86_64) as those are
+  // being defined.
+  if (Triple.getSubArch() == llvm::Triple::NoSubArch ||
+      Triple.getSubArch() == llvm::Triple::SPIRSubArch_fpga) {
```

Comment:
To clarify, does this have to do with ocloc/opencl-aot's CLI requiring a separate flag for build options?

Reply:
Yes, -g is passed to both host and device FE calls.

As for your second question, the separate flag for build options is part of the reason why I'm only doing this for device and fpga only.  It is the case, but they haven't solidified on behaviors yet so passing them now is pre-mature.  A new change will need to be done when those targets are ready.